src/CoreWCF.Primitives/tests/ServiceBuilderTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using CoreWCF.Channels;
+using CoreWCF.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace CoreWCF.Primitives.Tests
+{
+    public class ServiceBuilderTests
+    {
+        [Fact]
+        public void ServiceBuilderCanChain()
+        {
+            var services = new ServiceCollection();
+            services.AddServiceModelServices();
+            var serviceProvider = services.BuildServiceProvider();
+            var builder = new ServiceBuilder(serviceProvider);
```

Comment:
I'm not a fan of having product code referencing a test assembly with an InternalsVisibleTo attribute. I believe that if one assembly is strong name signed, then the reference to the other one must be too. Once I get all the signing stuff working, this will break this test unless the tests are also signed.  
It looks like you only need this to be able to reference and construct an instance of ServiceBuilder. The AddServiceModelServices adds this to the service collection as well as a factory which returns the same singleton instance when requesting an IServiceBuilder. So you should be able to replace this line of code with the following and remove the InternalsVisibleTo attribute.
```suggestion
            var builder = serviceProvider.GetRequiredService<IServiceBuilder>();
```